### PR TITLE
Improve ACL Data and ATT support

### PR DESCRIPTION
**Pull request by danielspofford**
- ACL Data
  - Fix handle and flag parsing.
  - Add support for ATT.
    - PDUs.
      - New.
        - ReadByTypeReq.
        - ReadByTypeRsp.
    - ErrorCodes.
      - Fix typo in description for "Attribute Not Long".
- Commands.
  - Update support for ControllerAndbaseband.
    - New.
      - Flush.
      - HostBufferSize.
      - SetControllerToHostFlowControl.
      - SetEventMaskPage2.
      - WriteFlowControlMode.
    - Fixed.
      - SetEventMask encoded/decoded incorrectly.
  - Add support for InformationalParameters.
    - New.
      - ReadBdAddr
      - ReadBufferSize.
      - ReadLocalSupportedFeatures.
  - Update support for LEController.
    - New.
      - ReadLocalSupportedFeatures.
    - Fixed.
      - SetEventMask encoded/decoded incorrectly.
    - Improved.
      - SetAdvertisingEnable.
- Events.
  - New.
    - HardwareError.
    - NumberOfCompletedPackets.
  - Fixed.
    - ConnectionComplete and DisconnectionCompleted now encodes/decodes
      connection_handles correctly.

### CODE REVIEW

File: lib/harald/hci/commands/informational_parameters/read_local_supported_features.ex [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+        pause_encryption: pause_encryption,
+        afh_capable_master: afh_capable_master,
+        afh_classification_master: afh_classification_master,
+        enhanced_data_rate_esco_2_mb_s_mode: enhanced_data_rate_esco_2_mb_s_mode,
+        enhanced_data_rate_esco_3_mb_s_mode: enhanced_data_rate_esco_3_mb_s_mode,
+        three_slot_enhanced_data_rate_esco_packets: three_slot_enhanced_data_rate_esco_packets,
+        extended_inquiry_response: extended_inquiry_response,
+        simultaneous_le_and_br_edr_to_same_device_capable_controller:
+          simultaneous_le_and_br_edr_to_same_device_capable_controller,
+        resserved_50: resserved_50,
```
Comment (primis) [inline lib/harald/hci/commands/informational_parameters/read_local_supported_features.ex:135]: doulbe s

File: lib/harald/hci/commands/informational_parameters/read_local_supported_features.ex [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+            sniff_subrating: _,
+            pause_encryption: _,
+            afh_capable_master: _,
+            afh_classification_master: _,
+            enhanced_data_rate_esco_2_mb_s_mode: _,
+            enhanced_data_rate_esco_3_mb_s_mode: _,
+            three_slot_enhanced_data_rate_esco_packets: _,
+            extended_inquiry_response: _,
+            simultaneous_le_and_br_edr_to_same_device_capable_controller: _,
+            resserved_50: _,
```
Comment (primis) [inline lib/harald/hci/commands/informational_parameters/read_local_supported_features.ex:229]: Same as above -- double s

File: lib/harald/hci/events/number_of_completed_packets.ex [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+          {acc_handle, acc_num_completed}
+      end)
+
+    {:ok, <<num_handles, connection_handles::binary, host_num_completed_packets::binary>>}
+  end
+
+  @impl Event
+  def decode(<<num_handles, arrayed_data::binary>>) do
+    {the_map, remaining_bin} =
```
Comment (primis) [inline lib/harald/hci/events/number_of_completed_packets.ex:25]: Maybe rename "the_map" to something more descriptive

**Review by primis — COMMENTED** [review]
Just a Typo to fix before merge

File: lib/harald/hci/events/number_of_completed_packets.ex (hunk 3 above)
Comment (primis) [reply lib/harald/hci/events/number_of_completed_packets.ex:25]: Suggestion but feel free to override and keep it as is: data_map

**Review by primis — APPROVED** [review]

File: lib/harald/hci/events/number_of_completed_packets.ex (hunk 3 above)
Comment (danielspofford) [reply lib/harald/hci/events/number_of_completed_packets.ex]: They're handles, lets callem that :)

File: lib/harald/hci/commands/informational_parameters/read_local_supported_features.ex (hunk 2 above)
Comment (danielspofford) [reply lib/harald/hci/commands/informational_parameters/read_local_supported_features.ex]: 🙇

File: lib/harald/hci/commands/informational_parameters/read_local_supported_features.ex (hunk 1 above)
Comment (danielspofford) [reply lib/harald/hci/commands/informational_parameters/read_local_supported_features.ex]: 🙇

**Pull request merged**